stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/messages/ThreadParticipants.kt

```diff
@@ -1,44 +1,79 @@
 package io.getstream.chat.android.compose.ui.components.messages
 
-import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.size
-import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.zIndex
 import coil.compose.rememberImagePainter
 import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.compose.state.messages.MessageAlignment
 import io.getstream.chat.android.compose.ui.components.avatar.Avatar
 import io.getstream.chat.android.compose.ui.theme.ChatTheme
 
+private const val DEFAULT_PARTICIPANTS_LIMIT = 4
+
 /**
- * Shows a row of participants in the message thread, if they exist.
+ * Represents the last two participants in the thread.
  *
- * @param participants List of users in the thread.
+ * @param participants The list of participants in the thread.
+ * @param alignment The alignment of the parent message.
  * @param modifier Modifier for styling.
- * @param text Text of the label.
+ * @param participantsLimit The limit of the number of participants shown in the component.
  */
 @Composable
 public fun ThreadParticipants(
     participants: List<User>,
+    alignment: MessageAlignment,
     modifier: Modifier = Modifier,
-    text: String,
+    participantsLimit: Int = DEFAULT_PARTICIPANTS_LIMIT,
 ) {
-    Row(modifier = modifier.padding(start = 4.dp, end = 4.dp, top = 4.dp)) {
-        Text(
-            modifier = Modifier.padding(end = 4.dp),
-            text = text,
-            style = ChatTheme.typography.footnoteBold,
-            color = ChatTheme.colors.primaryAccent
-        )
+    Box(modifier) {
+        /**
+         * If we're aligning the message to the start, we just show items as they are, if we are showing them from the
+         * end, then we need to reverse the order.
+         */
+        /**
+         * If we're aligning the message to the start, we just show items as they are, if we are showing them from the
+         * end, then we need to reverse the order.
+         */
```

Comment:
It looks duplicated. 👯

Reply:
Good point - not sure what happened here!